### PR TITLE
Throw cmake error when setting MOCK ON without gcc

### PR DESCRIPTION
**Pull request by vhvb1989**
fixes: https://github.com/Azure/azure-sdk-for-c/issues/912

### CODE REVIEW

**Comment by CIPop** [comment]
Tested on Windows:

```
CMake Error at CMakeLists.txt:111 (message):
  Unsupported Compiler Option.

  Option `UNIT_TESTING_MOCK` is not supported on MSVC.  It is only supported
  by GNU gcc.

  Remove option `UNIT_TESTING_MOCK` to build tests without using MOCK
```

File: CMakeLists.txt [hunk 1]
```diff
@@ -102,3 +102,16 @@ if (UNIT_TESTING)
   # Storage
   add_subdirectory(sdk/tests/storage/blobs)
 endif()
+
+# Fail geneation when setting MOCKS ON without GCC
```
Comment (ahsonkhan) [inline CMakeLists.txt:106]:
```suggestion
# Fail generation when setting MOCKS ON without GCC
```

File: CMakeLists.txt [hunk 2]
```diff
@@ -102,3 +102,16 @@ if (UNIT_TESTING)
...
+if(UNIT_TESTING_MOCKS)
+  if(UNIT_TESTING)
+    if(NOT CMAKE_C_COMPILER_ID MATCHES "GNU")
+      # Fail generation when asking for MOCK withoyt GCC
```
Comment (ahsonkhan) [inline CMakeLists.txt:110]:
```suggestion
      # Fail generation when asking for MOCK without GCC
```

File: CMakeLists.txt [hunk 3]
```diff
@@ -102,3 +102,16 @@ if (UNIT_TESTING)
...
+      message(FATAL_ERROR "Unsupported Compiler Option.\nOption `UNIT_TESTING_MOCK` is not supported on ${CMAKE_C_COMPILER_ID}. It is only supported by GNU gcc.\nRemove option `UNIT_TESTING_MOCK` to build tests without using MOCK")
```
Comment (ahsonkhan) [inline CMakeLists.txt:111]: Isn't the option called `UNIT_TESTING_MOCKS` and not `UNIT_TESTING_MOCK`?
```suggestion
      message(FATAL_ERROR "Unsupported Compiler Option.\nOption `UNIT_TESTING_MOCKS` is not supported on ${CMAKE_C_COMPILER_ID}. It is only supported by GNU gcc.\nRemove option `UNIT_TESTING_MOCKS` to build tests without using MOCK")
```

File: CMakeLists.txt [hunk 4]
```diff
@@ -102,3 +102,16 @@ if (UNIT_TESTING)
...
+    endif()
+  else()
+    # Mention MOCK is ignored
+    message(WARNING "UNIT_TESTING_MOCK will be ignored because UNIT_TESTING is not enabled.")
```
Comment (ahsonkhan) [inline CMakeLists.txt:115]:
```suggestion
    message(WARNING "UNIT_TESTING_MOCKS will be ignored because UNIT_TESTING is not enabled.")
```

**Pull request merged**